eng/common/scripts/SemVer.ps1

```diff
@@ -360,6 +394,38 @@ class AzureEngSemanticVersion : IComparable {
       Write-Host "Error: Python beta string did not correctly increment"
     }
 
+    $versionString = "0.1.2"
+    $version = [AzureEngSemanticVersion]::ParseVersionString($versionString)
+    $version.IncrementAndSetToPrerelease()
+    $expected = "0.2.0"
+    if ($expected -ne $version.ToString()) {
+      Write-Host "Error: version string did not correctly increment. Expected: $expected, Actual: $version"
+    }
+
+    $version.IncrementAndSetToPrerelease("patch")
+    $expected = "0.2.1"
+    if ($expected -ne $version.ToString()) {
+      Write-Host "Error: version string did not correctly increment. Expected: $expected, Actual: $version"
+    }
+
+    $version.IncrementAndSetToPrerelease("minor")
+    $expected = "0.3.0"
+    if ($expected -ne $version.ToString()) {
+      Write-Host "Error: version string did not correctly increment. Expected: $expected, Actual: $version"
+    }
+
+    $version.IncrementAndSetToPrerelease("major")
+    $expected = "1.0.0-beta.1"
```

Comment:
Why would this be the expected? You might want to consider resetting or at least asserting what $version should be here if you plan to reuse the same object that is being mutated.

Reply:
I've updated the tests to be more explicit on the version before the increment call.

For that major bump, we do expect a "0.1.2" to become "1.0.0", but then append a prerelease tag.
So `"0.1.2" IncrementAndSetToPrerelease("major")` -> `1.0.0-beta.1`
But `"1.0.0-beta.1" IncrementAndSetToPrerelease("major")` -> `1.0.0-beta.2`

Reply:
The method name `IncrementPrerelease` would be more appropriate.   
  Prerelease tagged input only increments the prerelease number
  Release input increments the segment, then appends a prerelease tag if  `major != 0`
  
The resulting version is always prerelease.